doc/user-guides/secure-protect-connect.md

```diff
@@ -0,0 +1,372 @@
+
+```
+
+The limit here is artificially low in order for us to show it working easily. Lets test it with our endpoint:
+
+```
+for i in {1..10}; do curl -k --resolve api.${KUADRANT_ZONE_ROOT_DOMAIN}:443:${INGRESS_HOST} "https://api.$KUADRANT_ZONE_ROOT_DOMAIN/cars" && sleep 1; done
+```
+
+Here we should see `409s` start returning after the 5th request.
+
+### Define a Gateway `AuthPolicy`
+
+So communication is secured and we have some protection for our infrastructure, but we do not trust any client to access our endpoints. By default we want to allow only authenticated access. To protect our gateway we will add a `DENY ALL` `AuthPolicy`. Later we will override this with a specific `AuthPolicy` for the API.
+
+```
+kubectl --context kind-kuadrant-local apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta2
+kind: AuthPolicy
+metadata:
+  name: deny-all
+  namespace: kuadrant-system
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: api-gateway
+  rules:
+    authorization:
+      deny-all:
+        opa:
+          rego: "allow = false"
+    response:
+      unauthorized:
+        headers:
+          "content-type":
+            value: application/json
+        body:
+          value: |
+            {
+              "error": "Forbidden",
+              "message": "Access denied by default by the gateway operator. If you are the administrator of the service, create a specific auth policy for the route."
+            }
+EOF
+```
+
+Lets test again. This time we expect a `403`
+
+```
+curl -k --resolve api.${KUADRANT_ZONE_ROOT_DOMAIN}:443:${INGRESS_HOST}  "https://api.$KUADRANT_ZONE_ROOT_DOMAIN/cars"
+
+```
+
+### Define DNSPolicy 
+(skip of you did not setup a DNS provider during the setup)
+
+Now we have our gateway protected and communications secured, we are ready to configure DNS so it is easy for clients to connect and access the APIs we intend to expose via this gateway. Note during the setup of this walk through, we setup a `DNS Provider` secret and a `ManagedZone`.
+
+```
+kubectl --context kind-kuadrant-local apply -f - <<EOF
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: simple-dnspolicy
+  namespace: kuadrant-system
+spec:
+  routingStrategy: simple
+  targetRef:
+    name: api-gateway
+    group: gateway.networking.k8s.io
+    kind: Gateway
+EOF
+
+kubectl wait dnspolicy simple-dnspolicy -n kuadrant-system --for=condition=ready
+```
+
+If you want to see the DNSRecord created by the this policy, execute 
+
+```
+kubectl get dnsrecord api-gateway-api -n kuadrant-system -o=yaml
+```
+
+So now we have a wildcard DNS record to bring traffic to our gateway. 
+
+Lets test again. This time we expect a `403` still as the DENY_ALL is still in effect. Notice we no longer need to set the Host header directly.
+
+
+```
+curl -k --resolve api.${KUADRANT_ZONE_ROOT_DOMAIN}:443:${INGRESS_HOST}  "https://api.$KUADRANT_ZONE_ROOT_DOMAIN/cars" -i
```

Comment:
Do we still need `--resolve` after creating the DNSPolicy?

Reply:
yeah we shouldn't need that actually